lychee-lib/src/extract.rs

```diff
@@ -224,11 +233,17 @@ mod test {
     #[test]
     fn test_file_type() {
         // FIXME: Assume plaintext in case a path has no extension
+        // REVIEW: can we just fix this or is there some compatablility
+        // reason we can't ?
         // assert_eq!(FileType::from(Path::new("/")), FileType::Plaintext);
+        // assert_eq!(FileType::from("test"), FileType::Plaintext);
         assert_eq!(FileType::from("test.md"), FileType::Markdown);
+        assert_eq!(FileType::from("TEST.MD"), FileType::Markdown);
```

Comment:
-1
On modern file system it's case sensitive.
According to mimeinfo of `text/markdown`, valid extension is one of `.md` `.mkd` and `.markdown`.

If a `.MD` extension is recognized as markdown, it IS A BUG.

Reply:
Are you saying that handling an html file incorrectly because it is called "index.HTML" is okay? Or are you only referring to Markdown files? In that case I guess If we do what I proposed in my comment above we can only check if certain extension match insensitively, I don't see how handling files with extensions that are not _technically_ in spec is an improvement

Reply:
I'm not talking about the REVIEW comment.

What I'm taking is the additional `assert_eq` tests you added are not sound. They should be removed.

Reply:
Ah, that makes sense, are you okay with changing that to a ".HTML" test or are uppercase extension not technically in-spec in general?

Reply:
To incorporate them is very complicated.
For modern Linux/Unix/FreeBSD systems, the file system should be case-sensitive. Thus these are unknown schemes and are recognized as planttext. (But newer Linux kernel supports opt-in case-insensitive feature)
For macOS, HFS/HFS+ defaults to case-insensitive if operating system is installed before 10.3. Otherwise it's case-sensitive. For APS, it's case-sensitive.
For Windows, NTFS is actually case sensitive internally, but defaults to case-insensitive in user space.

If you really want to incorporate those tests, you need conditional compilations based on operating system and whehther the filesystem is case sensitive.

My opinion is these cases are too complicated to handle. Just remove those tests.